crates/context/src/journal_init.rs

```diff
@@ -0,0 +1,21 @@
+use super::journaled_state::JournaledState;
+use database_interface::EmptyDB;
+
+/// A clonable version of JournaledState that uses EmptyDB.
+pub type JournalInit = JournaledState<EmptyDB>;
+
+impl<DB> From<&JournaledState<DB>> for JournalInit {
+    fn from(state: &JournaledState<DB>) -> Self {
```

Comment:
This is an expensive operation that clones a lot of fields. would remove it and just add `into_init()` and `to_init()` functions to `JournaledState`

Reply:
thx added that! so to_init() is still cloning similar to the rust from/into convention. I think that is what you wanted?